core/src/main/kotlin/com/tschuchort/compiletesting/KotlinCompilation.kt

```diff
@@ -259,8 +259,11 @@ class KotlinCompilation : AbstractKotlinCompilation<K2JVMCompilerArguments>() {
 		val messages: String
 	) {
 		/** class loader to load the compile classes */
-		val classLoader = URLClassLoader(arrayOf(outputDirectory.toURI().toURL()),
-			this::class.java.classLoader)
+		val classLoader = URLClassLoader(
+			// Include the original classpaths and the output directory to be able to load classes from dependencies.
+			classpaths.plus(outputDirectory).map { it.toURI().toURL() }.toTypedArray(),
+			this::class.java.classLoader
+		)
```

Comment:
Before I merge this I have just one question: Do you think there could be problems if class files appear multiple times? In many cases the `classpaths` will contain the same classes as the parent classloader `this::class.java.classLoader`.

Reply:
No, I don't believe so. I'd expect them to be identical so it doesn't make a difference. Class loaders load the first class and don't attempt to load the 2nd class to my knowledge. (Android is different and complains about duplicated classes, but that's irrelevant for this library)